test/FunctionalTests/E2ETestBase.cs

```diff
@@ -141,5 +142,19 @@ protected virtual void AssertResponse(HttpResponseMessage response, string respo
         {
             throw new NotImplementedException("Must be overriden in derived types.");
         }
+
+        private static int GetNextPort()
```

Comment:
Yes copied. Probably not worth maintaining as an API. It's simple enough and it's just test code.